EventPlanner/app/src/main/java/m3/eventplanner/viewmodel/HomeScreenViewModel.java

```diff
@@ -0,0 +1,274 @@
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+import m3.eventplanner.clients.ClientUtils;
+import m3.eventplanner.models.GetEventDTO;
+import m3.eventplanner.models.GetEventTypeDTO;
+import m3.eventplanner.models.GetOfferingDTO;
+import m3.eventplanner.models.PagedResponse;
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+
+public class HomeScreenViewModel extends ViewModel {
+
+    private final ClientUtils clientUtils;
+
+    private final MutableLiveData<List<GetEventDTO>> topEvents = new MutableLiveData<>();
+    private final MutableLiveData<List<GetOfferingDTO>> topOfferings = new MutableLiveData<>();
+
+    private final MutableLiveData<PagedResponse<GetEventDTO>> pagedEvents = new MutableLiveData<>();
+    private final MutableLiveData<PagedResponse<GetOfferingDTO>> pagedOfferings = new MutableLiveData<>();
+    private final MutableLiveData<List<GetEventTypeDTO>> eventTypes = new MutableLiveData<>();
+
+
+    private int currentEventPage = 0;
+    private final int eventPageSize = 3;
+    private int totalEventPages = 0;
+    private int totalEvents = 0;
+
+    private int currentOfferingPage = 0;
+    private final int offeringPageSize = 3;
+    private int totalOfferingPages = 0;
+    private int totalOfferings = 0;
+    private enum PaginationContext {
+        EVENTS, OFFERINGS
+    }
+
+    private PaginationContext currentPaginationContext = PaginationContext.EVENTS;
+
+    private Integer filterEventTypeId;
+    private String filterEventLocation;
+    private Integer filterEventMaxParticipants;
+    private Double filterEventMinRating;
+    private String filterEventStartDate;
+    private String filterEventEndDate;
+    private String eventSearchQuery;
+    private String sortEventsBy;
+    private String sortEventsDirection;
+
+
+    public HomeScreenViewModel(ClientUtils clientUtils) {
+        this.clientUtils = clientUtils;
+    }
+    public LiveData<List<GetEventDTO>> getTopEvents() {
+        return topEvents;
+    }
+
+    public LiveData<List<GetOfferingDTO>> getTopOfferings() {
+        return topOfferings;
+    }
+
+    public LiveData<PagedResponse<GetEventDTO>> getPagedEvents() {
+        return pagedEvents;
+    }
+
+    public LiveData<PagedResponse<GetOfferingDTO>> getPagedOfferings() {
+        return pagedOfferings;
+    }
+
+    public void loadTopEvents() {
+        clientUtils.getEventService().getTopEvents().enqueue(new Callback<Collection<GetEventDTO>>() {
+            @Override
+            public void onResponse(Call<Collection<GetEventDTO>> call, Response<Collection<GetEventDTO>> response) {
+                if (response.isSuccessful() && response.body() != null && response.body() != null) {
+                    if (response.body().isEmpty()) {
+                        topEvents.setValue(null);
+                    } else {
+                        topEvents.setValue(new ArrayList<>(response.body()));
```

Comment:
if (response.isSuccessful() && response.body() != null) {
    topOfferings.setValue(new ArrayList<>(response.body()));
}
should be just fine